reflex/event.py

```diff
@@ -652,6 +655,58 @@ def _callback_arg_spec(eval_result):
     return [eval_result]
 
 
+def set_http_only_cookie(
```

Comment:
i think it would be ideal to handle these kinds of cookies in the same way we handle `rx.Cookie` currently, instead of having a kind of separate parallel API.

In `state.py`, we can add the `http_only` field to the existing `Cookie` class.

Then in `state.js` if the `http_only` flag is set, instead of setting the cookie via JS DOM, we set it using the `fetch` mechanism. Right about here: https://github.com/reflex-dev/reflex/blob/fb2c3606d951aab77e4f8b61a9d97ddc43a6a87d/reflex/.templates/web/utils/state.js#L528

The last piece of the puzzle is applying any cookies that get set when the websocket connects to the state so they can be easily accessed... i'll have to think about that one a little more to give a proper recommendation. so accessing them through the new `cookie` field in `HeaderData` is probably fine for now.